### DISCUSSION TITLE
[Beta Projects] Trigger issue templates when adding new issue

### DISCUSSION THREAD

**Discussion by jess-williamson-treespleasegames** [community/community#6481] [Projects and Issues] [answered]
As a user I want to be able to add new items to my project backlog and ensure that the correct issue template type is used.

**Comment by cirelli94** [comment]
I do agree templates are needed as in the normal issues creating!

**Comment by kevinwhoffman** [comment]
## Current outcome

My team uses issue templates for every issue, which means I can't use the issue creation feature on the project board until it supports templates.

## Desired outcome

When adding a new draft issue using the plus button at the bottom of the project board, I want to select an issue template which then pre-populates the issue title and content. This option could be presented right below the current option to "Add an issue from a repository":

![Google Chrome-@kevinwhoffman's untitled project-20220107T235200](https://user-images.githubusercontent.com/1938671/148621703-d3c1500d-def4-40e8-bc52-1e14cb6ff1b5.png)

### Which issue templates?

Since project boards can span multiple repositories, GitHub currently has no way of knowing which set of issue templates to display when you create a new draft issue. It is not until you select "Convert to issue" that GitHub actually associates the issue with a specific repository. Perhaps there could be a setting in the project that allows you to select which repository to reference when loading issue templates.

**Comment by tom-studiohakuna** [comment]
+1

Relates to:
https://github.com/github/feedback/discussions/5804
https://github.com/github/feedback/discussions/10303

**Comment by cirelli94** [comment]
> It is not until you select "Convert to issue" that GitHub actually associates the issue with a specific repository.

Perhaps there could be a setting in the project that allows you to select which repository to reference when loading issue templates.
Or perhaps it could insert the template in the body of the issue and the title of the issue be appended to the default title of the template.

**Comment by vicdotdevelop** [comment]
+1 I love writing issue by issue in my backlog - unfortunatley it does not support the templates that I have created in the main repo of my organization which belongs to the project

**Comment by jinwoong16** [comment]
Because the projects is much more convenient, it is too sad that I cannot choose the issues templates. 😢

**Comment by mbaler** [comment]
Bump on this!! 🙏

**Comment by davinc** [comment]
This is a required feature, 

1. We have a common project for various departments to work on
2. Various departments have their own repositories for issue management with templates defined for work
3. While we create roadmap in project with items, there is not way to convert these items to specific issue templates after selecting a repository.
4. Perhaps we can define a default issue template in repository when issue is triggered from a project?
5. Is there a automated way of achieving this using github actions?
6. At least allow issue template to be applied after the blank issue is created from a draft item.

**Comment by christrunk** [comment]
+1 on this please 🥇

**Comment by rileybroughten** [comment, marked as answer]
👋 You can now use an issue templates when creating a new issue directly from a project! Check out the [changelog](https://github.blog/changelog/2023-07-13-github-issues-projects-july-13th-update/#heavy_plus_sign-accessing-issue-templates-in-projects) announcement for more details 😁

**Reply by vicdotdevelop** [reply]
<img width="743" alt="image" src="https://github.com/community/community/assets/60963842/a9514c45-361c-4c4d-b577-71eee4198489">

works like a charm @rileybroughten  🚀

**Reply by JackEAllen** [reply]
@rileybroughten  A little late to this thread, but is there any chance we could have this for draft issues in the future too? My team would find it really useful to be able to create draft issues within a template when creating a new item from a project board or table. 

Kind of like this:

<img width="765" alt="image" src="https://github.com/community/community/assets/32413847/ad5dc70e-dd0d-4364-bf85-e23f53c51fa4">


Often requirements gathering can take time and it makes sense to write down in a draft, but not a fully fledged issue, I find it cumbersome copying in template markdown manually as a work around to this.

It looks like a few others have had the same thought:
- #50026 : https://github.com/orgs/community/discussions/50026#discussioncomment-5416233
- #14010

**Reply by joaomas** [reply]
> @rileybroughten A little late to this thread, but is there any chance we could have this for draft issues in the future too? My team would find it really useful to be able to create draft issues within a template when creating a new item from a project board or table.
> 
> Kind of like this:
> <img alt="image" width="765" src="https://private-user-images.githubusercontent.com/32413847/330736932-ad5dc70e-dd0d-4364-bf85-e23f53c51fa4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NDQ4MzEsIm5iZiI6MTcyNzc0NDUzMSwicGF0aCI6Ii8zMjQxMzg0Ny8zMzA3MzY5MzItYWQ1ZGM3MGUtZGQwZC00MzY0LWJmODUtZTIzZjUzYzUxZmE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDAxMDIxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhMTQ3YjdmMjM0Y2RhM2M0MGUwMzc1MWY5ZTk1ZmZmYWM2N2FkYmM4N2ZjODllNjgzYzkyZjZkNTNjZjFjNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.U-gL_hazPsTXjQvfAM62lkclRCfdAt5jrVhNgtLChS4">
> 
> Often requirements gathering can take time and it makes sense to write down in a draft, but not a fully fledged issue, I find it cumbersome copying in template markdown manually as a work around to this.
> 
> It looks like a few others have had the same thought:
> 
>     * [Templates for draft issues #50026](https://github.com/orgs/community/discussions/50026) : https://github.com/orgs/community/discussions/50026#discussioncomment-5416233
> 
>     * [Convert to issue does not use a template #14010](https://github.com/orgs/community/discussions/14010)

Yes! Will be very userfull.

**Comment by LaurentMag** [comment]
Hello, is it also possible to load a template when a draft is "convert to issue" ? i do not seem to see this option if it exist.

Thanks !

**Reply by rileybroughten** [reply]
It is not yet supported, but it is on our backlog!

**Reply by jehtee** [reply]
Yes please.  Creating issues should go through the same pipeline regardless of how it was initiated.  I can't wait for this feature.

**Reply by davinc** [reply]
yes please, this is required.

**Reply by rohitgadhiya** [reply]
yup. i've been wanting this for very long now.

**Reply by ashwinee-datar** [reply]
yes , this is required.

**Reply by aanalshahh** [reply]
yes, we do  need this feature

**Reply by njjoshi07** [reply]
yes , this is required.

**Reply by joaomas** [reply]
> Hello, is it also possible to load a template when a draft is "convert to issue" ? i do not seem to see this option if it exist.
> 
> Thanks !

Yes! Will be very userfull!!!!!!!!!!!!

**Reply by MarkOSullivan94** [reply]
This is what I was searching for
